plugins/wasm-rust/src/plugin_wrapper.rs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2023 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+use proxy_wasm::traits::{Context, HttpContext, RootContext};
+use proxy_wasm::types::{Action, Bytes};
+use serde::de::DeserializeOwned;
+use multimap::MultiMap;
+use crate::rule_matcher::SharedRuleMatcher;
+
+pub trait RootContextWrapper<PluginConfig> : RootContext
+where
+    PluginConfig: Default + DeserializeOwned + 'static + Clone,
+{
+
+    // fn create_http_context(&self, _context_id: u32) -> Option<Box<dyn HttpContext>> {
+    fn create_http_context_use_wrapper(&self, _context_id: u32) -> Option<Box<dyn HttpContext>> {
+        // trait 继承没法重写 RootContext 的 create_http_context，先写个函数让上层调下吧
```

Comment:
I might need to look at this design and see if there is a better way.

Reply:
I just started learning Rust. I don't know how to override the inherited trait function. The expected result is:
`
    fn create_http_context(&self, _context_id: u32) -> Option<Box<dyn HttpContext>> {
        match self.create_http_context_wrapper(_context_id) {
            Some(http_context) => Some(Box::new(PluginHttpWrapper::new(
                self.rule_matcher(),
                http_context,
            ))),
            None => None,
        }
    }
`

Reply:
Rust has generic specialization, but it is still in an unstable state.  
It is also possible to not rely on RootContext but rather treat RootContext as a field.

Since there aren't many Rust plugins available right now, we can merge your modifications first.